### PR TITLE
Update Lint to default to Puppet Approved criteria

### PR DESCRIPTION
**Pull request by cyberious**

### CODE REVIEW

File: lib/puppetlabs_spec_helper/rake_tasks.rb [hunk 1]
```diff
@@ -193,14 +193,21 @@ def revision(scm, target, ref)
...
-  PuppetLint.configuration.relative = true
-  PuppetLint.configuration.disable_class_inherits_from_params_class
-  PuppetLint.configuration.ignore_paths ||= []
-  PuppetLint.configuration.ignore_paths << "spec/fixtures/**/*.pp"
-  PuppetLint.configuration.ignore_paths << "pkg/**/*.pp"
+require 'puppet-lint/tasks/puppet-lint'
+# Must clear as it will not override the existing puppet-lint rake task since we require to import for
+# the PuppetLint::RakeTask
+Rake::Task[:lint].clear
+# Relative is not able to be set within the context of PuppetLint::RakeTask
```
Comment (domcleal) [inline lib/puppetlabs_spec_helper/rake_tasks.rb]: For the record, this is fixed in https://github.com/rodjek/puppet-lint/commit/c033c3739599161b8164e6d9738d0e0b3c8f1d72 (to be 1.1.1).

File: lib/puppetlabs_spec_helper/rake_tasks.rb [hunk 2]
```diff
@@ -193,14 +193,21 @@ def revision(scm, target, ref)
...
-  require 'puppet-lint/tasks/puppet-lint'
-  PuppetLint.configuration.relative = true
-  PuppetLint.configuration.disable_class_inherits_from_params_class
-  PuppetLint.configuration.ignore_paths ||= []
-  PuppetLint.configuration.ignore_paths << "spec/fixtures/**/*.pp"
-  PuppetLint.configuration.ignore_paths << "pkg/**/*.pp"
+require 'puppet-lint/tasks/puppet-lint'
+# Must clear as it will not override the existing puppet-lint rake task since we require to import for
+# the PuppetLint::RakeTask
+Rake::Task[:lint].clear
```
Comment (domcleal) [inline lib/puppetlabs_spec_helper/rake_tasks.rb]: also fixed in 1.1.1

**Comment by domcleal** [comment]
The minimum version of puppet-lint needs setting to >= 1.1.0 in the gemspec, as older versions don't have this ability IIRC.

**Comment by cyberious** [comment]
@domcleal so as I understand it, these are fixed but not released yet and don't know when they will be released?

**Comment by domcleal** [comment]
@cyberious yep

**Comment by tphoney** [comment]
This has went quiet, it also needs a rebase.

**Comment by cyberious** [comment]
@domcleal 4 months later still no Puppet-lint 1.1.1, @tphoney rebased

**Pull request merged**